src/languages/delphi.js

```diff
@@ -164,20 +164,28 @@ export default function(hljs) {
     relevance: 0,
     // Source: https://www.freepascal.org/docs-html/ref/refse6.html
     variants: [
+      {
+        // Regular numbers, e.g., 123, 123.456.
+        begin: /\b\d[\d_]*(\.\d[\d_]*)?/ },
       {
         // Hexadecimal notation, e.g., $7F.
-        begin: '\\$[0-9A-Fa-f]+' },
+        begin: /\$[\dA-Fa-f_]*/ },
       {
         // Octal notation, e.g., &42.
-        begin: '&[0-7]+' },
+        begin: /&[0-7][0-7_]*/ },
       {
         // Binary notation, e.g., %1010.
-        begin: '%[01]+' }
+        begin: /%[01_]*/ }
```

Comment:
This is not comparable because it can match no digits at all.

Reply:
Hi @joshgoebel,
This was an intentional change in [5d3110c](https://github.com/highlightjs/highlight.js/pull/3932/commits/5d3110c91f0869825f584d6645383de8a34707b1).
Delphi allows all digits to be omitted in binaryliterals.